distribution/registry.go

```diff
@@ -123,9 +124,24 @@ func newRepository(
 			Class:      repoInfo.Class,
 		}
 
+		var newAuthTransport = authTransport
+		//TODO: Support multiple challenges for loading TLS configs
+		if len(realmHost) > 0 && registryResolver != nil {
+			//trying to load a new TLS config for the auth challenge
```

Comment:
Same as above (although I think we could skip this comment, as it basically says the same as the debug log below.